test/test_Quaternion.jl

```diff
@@ -138,4 +138,13 @@ for _ in 1:100
         @test q ⊗ slerp(q1, q2, t) ≈ slerp(q ⊗ q1, q ⊗ q2, t)
         @test q ⊗ linpol(q1, q2, t) ≈ linpol(q ⊗ q1, q ⊗ q2, t)
     end
+    let # test matrix exponential
+        z = zeros(Quaternion{Float64}, 3, 3)
+        id = Matrix{Quaternion{Float64}}(I, 3, 3)
+        d = [Quaternion(randn(4)...) for i in 1:3]
+        expd = exp.(d)
+        D = exp(Array(Diagonal(d)))
```

Comment:
Can you include a test for a dense matrix also? Perhaps you could check it using `evalpoly`.